app/routers/google_connect.py

```diff
@@ -0,0 +1,261 @@
+    user = session.query(User).filter_by(id=1).first()
+    return user
+
+
+def is_client_secret_none() -> bool:
+    return CLIENT_SECRET_FILE is None
+
+
+def get_current_year_events(
+        credentials: Credentials, user: User,
+        session: SessionLocal = Depends()) -> list:
+    '''Getting user events from google calendar'''
+
+    currrnt_year = datetime.now().year
+    start = datetime(currrnt_year, 1, 1).isoformat() + 'Z'
+    end = datetime(currrnt_year + 1, 1, 1).isoformat() + 'Z'
+
+    service = build('calendar', 'v3', credentials=credentials)
+    events_result = service.events().list(
+        calendarId='primary',
+        timeMin=start,
+        timeMax=end,
+        singleEvents=True,
+        orderBy='startTime'
+    ).execute()
+
+    events = events_result.get('items', [])
+    return events
+
+
+def push_events_to_db(events: list, user: User,
+                      session: SessionLocal = Depends()) -> bool:
+    '''Adding google events to db'''
+    db_cleanup(user, session)
+
+    for event in events:
+        # Running over the events that have come from the API
+        location = None
+        title = event['summary']  # The Google event title
+
+        # support for all day events
+        if 'dateTime' in event['start'].keys():
+            # This case handles part time events (not all day events)
+            start = datetime.fromisoformat(event['start']['dateTime'])
+            end = datetime.fromisoformat(event['end']['dateTime'])
+        else:
+            # This case handles all day events
+            # It takes the date string in this format yyyy-mm-dd
+            # - according to the documentation
+            # and then split it and passing it to datetime constructor.
+            start_in_str = event['start']['date'].split('-')
+            start = datetime(
+                year=int(start_in_str[0]),
+                month=int(start_in_str[1]),
+                day=int(start_in_str[2])
+            )
+
+            end_in_str = event['end']['date'].split('-')
+            end = datetime(
+                year=int(end_in_str[0]),
+                month=int(end_in_str[1]),
+                day=int(end_in_str[2])
+            )
+
+        if 'location' in event.keys():
+            # This case handles if Google Event has a location attached
+            location = event['location']
+
+        create_google_event(title, start, end, user, location, session)
+    return True
+
+
+def create_google_event(title: str, start: datetime,
+                        end: datetime, user: User, location: str,
+                        session: SessionLocal = Depends()) -> Event:
+    return create_event(
+        # creating an event obj and pushing it into the db
+        db=session,
+        title=title,
+        start=start,
+        end=end,
+        owner_id=user.id,
+        location=location,
+        is_google_event=True
+    )
+
+
+def db_cleanup(user: User, session: SessionLocal = Depends()) -> bool:
```

Comment:
It's hard to tell what's the difference from `clean_up_old_credentials_from_db` when only looking at the name

Reply:
Fixed.